### PR TITLE
Html sortable table with styling for light and dark mode.

### PR DESCRIPTION
**Pull request by straburzynski**
Light mode:
image

Dark mode:
image

### CODE REVIEW

**Review by platan — CHANGES_REQUESTED** [review]
Proposed solution looks nice and works properly. It contains a lot of javascript (sorting in table) and styling. 
Sorting is most valuable for "duration". I don't think we really need an option to reverse order of rows sorted by duration or sort by other columns. It's a lot of javascript code which is not tested. It have to be maintained in the future. I suggest to:
- remove option to sort dynamically - sort only in HTML (preferred)
- alternatively use existing library for sorting tables (like https://gridjs.io/docs/examples/sorting, https://gridjs.io/docs/hello-world/#browser)

Regarding styling. I looks great, but this code have to be maintained in the future. Any change in styling of mermaid themes will lead to inconsistency between chart/table styling. I think simpler styling (not dependent on mermaid colors) should be easier to maintain.

File: src/main/kotlin/com/github/platan/tests_execution_chart/reporters/html/HtmlTableFormatter.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import com.github.platan.tests_execution_chart.TestExecutionScheduleReport
+
+class HtmlTableFormatter {
+
+    fun format(report: TestExecutionScheduleReport): String {
+        val tbody = StringBuilder()
+        report.results.sortedByDescending { it.durationMs }.map {
+            tbody.append(
+                """<tr><td>${it.testName}</td>
```
Comment (platan) [inline src/main/kotlin/com/github/platan/tests_execution_chart/reporters/html/HtmlTableFormatter.kt:11]: HTML special characters in testName/className values should be escaped. Currently test name `def "test 1 <tr>"() {` results with broken table:
Zrzut ekranu 2022-10-24 o 19 00 31

**Review by platan — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/github/platan/tests_execution_chart/reporters/html/HtmlTableFormatter.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                   <td>${it.durationMs}</td></tr>"""
+            )
+        }
+        return """<table id="table">
+            <thead>
+                <tr>
+                    <td>Test name</td>
+                    <td>Class name</td>
+                    <td>Result</td>
+                    <td class="selected order-desc">Duration (ms)</td>
```
[the remark this answers is not in the document]
Comment (straburzynski) [reply src/main/kotlin/com/github/platan/tests_execution_chart/reporters/html/HtmlTableFormatter.kt]: Yes, I've missed that, thanks - removed.

File: src/main/resources/template.html [hunk 3]
```diff
@@ -5,17 +5,56 @@
...
+                background-color: #191919;
+            }
         }
     </style>
     <script src="@MERMAID_SRC@"></script>
 </head>
 
 <body>
 <div id="chart-container"></div>
+<div id="table-container">@TABLE_SRC@</div>
```
[the remark this answers is not in the document]
Comment (straburzynski) [reply src/main/resources/template.html]: Changed.

File: src/main/resources/template.html [hunk 4]
```diff
@@ -29,5 +68,11 @@
     })
 </script>
 </body>
-
+<style>
+    @media (prefers-color-scheme: dark) {
+        #chart .taskText {
```
[the remark this answers is not in the document]
Comment (straburzynski) [reply src/main/resources/template.html:73]: It makes font color better more readable on dark mode, the default one is white and does not look good on light blue background:

image

File: src/main/kotlin/com/github/platan/tests_execution_chart/reporters/html/HtmlTableFormatter.kt (hunk 1 above)
Comment (straburzynski) [reply src/main/kotlin/com/github/platan/tests_execution_chart/reporters/html/HtmlTableFormatter.kt]: I've added escaping for html and test for checking </> characters.

Comment (platan) [reply src/main/kotlin/com/github/platan/tests_execution_chart/reporters/html/HtmlTableFormatter.kt]: Thanks for fixing and adding test for this case.

**Review by platan — APPROVED** [review]
Thank you for this feature!

**Pull request merged**